graftm/taxonomy.py

```diff
@@ -0,0 +1,176 @@
+# the Free Software Foundation, either version 3 of the License, or           #
+# (at your option) any later version.                                         #
+#                                                                             #
+# This program is distributed in the hope that it will be useful,             #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of              #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                #
+# GNU General Public License for more details.                                #
+#                                                                             #
+# You should have received a copy of the GNU General Public License           #
+# along with this program. If not, see <http://www.gnu.org/licenses/>.        #
+#                                                                             #
+###############################################################################
+__author__ = "Joel Boyd, Ben Woodcroft"
+__copyright__ = "Copyright 2015"
+__credits__ = ["Joel Boyd"]
+__license__ = "GPL3"
+__version__ = "0.0.1"
+__maintainer__ = "Joel Boyd"
+__email__ = "joel.boyd near uq.net.au"
+__status__ = "Development"
+###############################################################################
+# System imports
+import logging 
+
+###############################################################################
+###############################################################################
+###############################################################################
+
+class MalformedGreenGenesFormattedFile(Exception):
+    pass
+
+#-------------------------------------------------------------------------------
+
+class ProgrammingError(Exception):
+    pass
+
+#-------------------------------------------------------------------------------
+
+class MissingTaxonomy(Exception):
+    pass
+
+#-------------------------------------------------------------------------------
+
+class Taxonomy:
+
+    TAB='\t'
+    COM=','
+    GG_SEP_0='; '
+    GG_SEP_1=';'
+    
+    def __init__(self):
+        self.taxonomy = {}
+        
+    def _ammend_taxonomy_gaps(self):
+        '''
+        Checks that each value in the self.taxonomy dict has consistent
+        length. 
+        '''
+        try:
+            self.max = max([len(x) for x in self.taxonomy.values()])
+            for taxa_id, taxa_taxonomy_list in self.taxonomy.iteritems():
+                
+                if len(taxa_taxonomy_list) == self.max:
+                    continue
+                else:
+                    ammended_split_taxonomy = \
+                                taxa_taxonomy_list + \
+                                [''] * (self.max - len(taxa_taxonomy_list))
+                self.taxonomy[taxa_id] = ammended_split_taxonomy
+        except:
+            logging.error("Controlled exit from taxonomy.Taxonomy \
```

Comment:
Originally left them in so the error was captured in the .log file output, because I'm not sure if Exceptions are written to the file.

I've taken these out.